tests/unit/test_exec.py

```diff
@@ -0,0 +1,70 @@
+__copyright__ = 'Copyright (c) 2020-2021 Jina AI Limited. All rights reserved.'
+__license__ = 'Apache-2.0'
+
+import os
+from pathlib import Path
+
+import numpy as np
+import pytest
+from jina import Document, DocumentArray, Executor
+
+from executor import ImageHasher
+
+cur_dir = os.path.dirname(os.path.abspath(__file__))
+
+
+def test_config():
+    ex = Executor.load_config(str(Path(__file__).parents[2] / 'config.yml'))
+    assert ex.hash_type == 'perceptual'
+    assert ex.hash_size == 8
+
+
+def test_no_documents(hasher: ImageHasher):
+    docs = DocumentArray()
+    hasher.encode(docs=docs)
+    assert len(docs) == 0  # SUCCESS
+
+
+def test_docs_no_blobs(hasher: ImageHasher):
+    docs = DocumentArray([Document()])
+    hasher.encode(docs=docs)
+    assert len(docs) == 1
+    assert len(docs[0].chunks) == 0
+
+
+def test_embed_type(hasher, docs):
+    hasher.encode(docs=docs)
+    assert docs.embeddings.dtype == np.uint8
+
+    hasher_bool = ImageHasher(is_embed_bool=True)
+    hasher_bool.encode(docs=docs)
+    assert docs.embeddings.dtype == np.bool
+
+
+def test_contrast_match(hasher, docs_contrast):
+    hasher.encode(docs=docs_contrast)
+    docs_contrast.match(docs_contrast, metric='euclidean', use_scipy=True)
```

Comment:
Can you expand this test to several hashing types and assert equality on the same input pics?

Reply:
Is euclidean distance the correct one here?

Reply:
`Euclidean` and `Hamming` both works.